### PR TITLE
🐛 ClusterClass : fix propagate metadata to machines, KCP fix propagate annotations

### LINKED ISSUE #5168: ClusterClass: Topology metadata is not propagated to machines
**What steps did you take and what happened:**
[A clear and concise description on how to REPRODUCE the bug.]

I configured labels and annotations in the topology (controlPlane & workers) and they were only propagated to the metadata of MachineDeployments, ControlPlanes, InfrastructureMachineTemplates and BoostrapTemplates.

**What did you expect to happen:**

I expected that labels and annotations set in the topology (controlPlane & workers) are propagated to the machines as well.

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

We should also set the labels and annotations in the respective fields in ControlPlane (as defined in the contract) and MachineDeployments so that they are propagated to machines.

**Environment:**

- Cluster-api version:
- Minikube/KIND version:
- Kubernetes version: (use `kubectl version`):
- OS (e.g. from `/etc/os-release`):

/kind bug
/area topology
[One or more /area label. See https://github.com/kubernetes-sigs/cluster-api/labels?q=area for the list of labels]

### PR DESCRIPTION
**Pull request by sbueringer**
Signed-off-by: Stefan Büringer buringerst@vmware.com
Co-authored-by: fabriziopandini <fpandini@vmware.com>

**What this PR does / why we need it**:
WIth this PR metadata is propagated to control plane and MachineDeployment machines.

There were also two bugs in KCP:
* the webhook didn't allow changes to metadata
* KCP didn't propagate annotations

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #5168

### CODE REVIEW

**Comment by fabriziopandini** [comment]
/assign @vincepri 
This fixes label/annotation propagation from ClusterClass/Topology to ControlPlane and MachineDeployments.
TL;DR; we are propagating labels only to machines, not to ControlPlane, MachineDeployments and templates objects.

**Review by vincepri — APPROVED** [review]
/approve
/lgtm

**Comment by fabriziopandini** [comment]
/hold
for squash
